pyuvm/s21_pyuvm_reg_map.py

```diff
@@ -0,0 +1,162 @@
+## Import Main Packages
+from pyuvm import uvm_object, uvm_sequencer
+from pyuvm.s24_pyuvm_reg_includes import *
+from pyuvm.s25_pyuvm_adapter import *
+from pyuvm.s26_pyuvm_predictor import *
+from pyuvm.s18_pyuvm_reg_block import *
+
+# Class declaration: uvm_reg_map
+class pyuvm_reg_map(uvm_object):
+    # Constructor
+    def __init__(self, name="pyuvm_reg_map"):
+        super().__init__(name)
+        self._parent        = None
+        self._base_addr     = None
+        self._regs          = {}
+        self.name           = name
+        self.header         = name + " -- "
+
+    # Function called by the REG_BLOCK create_map funcction
+    # the parent value here should be a pyuvm_reg_block instance type
+    def configure(self, parent, base_addr):
+        if(isinstance(parent, pyuvm_reg_block)):
+            self._parent    = parent
+        else:
+            UVMFatalError("pyuvm_reg_map -- configure -- parent should be pyuvm_reg_block type")
+        self._base_addr = base_addr
+        ## No support for Byte_Addressing nor for Byte_en TODO: add
+
+    # get_parent
+    def get_parent(self):
+        return self._parent
+
+    # get_base_addr
+    def get_base_addr(self):
+        return self._base_addr
+
+    # add_reg
+    def add_reg(self, reg, offset):
+        self._regs[offset] = reg
+
+    # get_registers
+    def get_registers(self):
+        return list(self._regs.values())
+
+    # get_reg_by_offset
+    def get_reg_by_offset(self, offset):
+        return self._regs[offset]
+
+    ## set_predictor
+    def set_predictor(self, predictor):
+        if isinstance(predictor, pyuvm_reg_predictor):
+            self.predictor = predictor
+        else:
+            error_out(self.header,"predictor should be type of uvm_reg_predictor")
+
+    ## get_predictor
+    def get_predictor(self):
+        if self.predictor == None:
+            error_out(self.header,"predictor Not set")
+        else:
+            return self.predictor
+
+    ## set_adapter
+    def set_adapter(self, adapter):
+        if isinstance(adapter, pyuvm_reg_adapter):
+            self.adapter = adapter
```

Comment:
@EngRaff92 , I've updated `s18_register_model.py` recently with changes to `uvm_reg_map`(https://github.com/pyuvm/pyuvm/pull/157). Would it be possible to refactor some methods from there ?

Reply:
Absolutely which methods do you wanna refactor ?

Reply:
some instance variables from `__init__()` and the following methods

- `__is_addr_aligned`
- `add_reg`
- `add_submap`
- `set_sequencer`
- `__is_map_valid`
- `get_submap_offset`
- `set_submap_offset`
- `set_base_addr`
- `reset`
- `get_root_map`
- `get_parent`
- `get_parent_map`
- `get_base_addr`
- `get_n_bytes`

most of the them are not conflicting except `add_reg`, `set_sequencer` I think

Reply:
Hey so basically since the initial RAL setup will only have one MAP I do believe that I will skip for the time being set_submap_offset and get_submap_offset is that ok for you ?

Reply:
Yes, makes sense. We could add it once the base RAL implementation is done

Reply:
@crteja and @EngRaff92 ,  I would like to let you take the lead on RAL. I have to prepare classes this month for school at the end of the month, but I am happy to handle simple tasks.

Reply:
Sure